cluster-autoscaler/cloudprovider/utils.go

```diff
@@ -125,3 +125,15 @@ func JoinStringMaps(items ...map[string]string) map[string]string {
 	}
 	return result
 }
+
+// FirstNotEmpty returns the first non-empty string from the input list.
+// If all strings are empty or no arguments are provided, it returns an empty string.
+func FirstNotEmpty(strs ...string) string {
```

Comment:
This function is quite trivial, I don't see a lot of value in putting it here. We do have a few "util" packages like this that are random bags of functionality, but I see this as an anti-pattern we should try to avoid. (See also: https://google.github.io/styleguide/go/best-practices#util-packages). Can we just put this function in the file that is using it? The alternative would be to have a more targeted package for it, though this would be an overkill for a simple function like this.

Reply:
@x13n moved it to alicloud_cloud_config.go (the using file)